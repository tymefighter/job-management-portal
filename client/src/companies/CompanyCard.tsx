import "../styles/CompanyCard.scss";

interface CompanyCardProps {
    imgUrl: string;
    name: string;
    description: string;
};

function CompanyCard({imgUrl, name, description}: CompanyCardProps) {
    return (
        <div className="company-card">
            <img className="company-card__img" src={imgUrl} alt={name} />
            <h2 className="company-card__name">{name}</h2>
            <p className="company-card__para">{description}</p>
        </div>
    );
}

export default CompanyCard;