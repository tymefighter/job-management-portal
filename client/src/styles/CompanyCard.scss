.company-card {
    display: flex;
    flex-direction: column;
    border: 0.2rem solid black;
}

.company-card__img {
    height: 25rem;
    width: 100%;
}

.company-card__name {
    font-size: 1.5rem;
    text-align: center;
}

.company-card__para {
    font-size: 1.2rem;
    padding: 0 0.5rem;
}