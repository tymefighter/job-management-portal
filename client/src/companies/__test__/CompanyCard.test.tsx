import { render } from "@testing-library/react";
import CompanyCard from "../CompanyCard";

test("test provided information in company card", () => {
    const imgUrl = "/this/is/an/image/url.png";
    const name = "This is the company's name";
    const description = "This is the company description";

    const element = (
        <CompanyCard 
            imgUrl={imgUrl} name={name} description={description}
        />
    );

    const {getByAltText, queryByText} = render(element);

    expect(getByAltText(name).getAttribute("src")).toEqual(imgUrl);
    expect(queryByText(name)).toBeTruthy();
    expect(queryByText(description)).toBeTruthy();
});

test("company card snapshot test", () => {
    const imgUrl = "/this/is/an/image/url.png";
    const name = "This is the company's name";
    const description = "This is the company description";

    const element = (
        <CompanyCard 
            imgUrl={imgUrl} name={name} description={description}
        />
    );

    const {container} = render(element);

    expect(container).toMatchSnapshot("CompanyCard Component snapshot");
});