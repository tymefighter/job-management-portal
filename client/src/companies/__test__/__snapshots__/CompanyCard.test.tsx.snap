// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`company card snapshot test: CompanyCard Component snapshot 1`] = `
<div>
  <div
    class="company-card"
  >
    <img
      alt="This is the company's name"
      class="company-card__img"
      src="/this/is/an/image/url.png"
    />
    <h2
      class="company-card__name"
    >
      This is the company's name
    </h2>
    <p
      class="company-card__para"
    >
      This is the company description
    </p>
  </div>
</div>
`;
